CONTRIBUTING.md

```diff
@@ -0,0 +1,254 @@
+Contributing
+============
+
+Thank you for contributing to vimperator-labs!
+
+Following are some guidelines designed to make the progress as smooth as
+possible. If you have any questions, feel free to drop by
+`#vimperator@freenode` or create an issue.
+
+Issues
+------
+
+A few things to keep in mind when creating a new issue:
+
+- Follow the issue template
+- Attach relevant configuration or RC file if applicable (e.g. your
+  `.vimperatorrc` or part of it).
+- Check if a fresh profile solves the issue
+    - `$ firefox -no-remote -P`
+- Confirm if it works without your configuration
+    - `$ firefox -no-remote -P <fresh profile> -vimperator "+u NONE"`
+
+Pull requests
+-------------
+
+- Title and commit message(s) should include relevant issue ID(s)
+- For any new or changed feature, AsciiDoc documentation and an entry in the
+  NEWS file is required for the patch to be accepted.
+
+Development installation
+------------------------
+
+Creating and installing a new XPI file after each update is cumbersome.
+Instead, create an extension linking to the vimperator directory, for example
+of a Git clone.
+
+1. Find the location your [Firefox profile](http://kb.mozillazine.org/Profile_folder_-_Firefox)
+    - E.g. `/home/user/.mozilla/firefox/<hash>.default`
+2. Go into the `extensions` directory
+3. Delete `vimperator@mozdev.org.xpi`
+4. Create a `vimperator@mozdev.org` text file
+    - Add the absolute path to your Vimperator directory
+    - E.g. `/home/user/code/vimperator-labs/vimperator`
+5. Restart Firefox
+
+Hacking
+-------
+
+If you've taken to hacking Vimperator source code, we hope that you'll share
+your changes. In case you do, please keep the following in mind, and we'll be
+happy to accept your patches.
+
+### Documentation
+
+First of all, all new features and all user-visible changes to existing
+features need to be documented. That means editing the appropriate help files
+and adding a NEWS entry where appropriate. When editing the NEWS file, you
+should add your change to the top of the list of changes. If your change alters
+an interface (key binding, command) and is likely to cause trouble, prefix it
+with `IMPORTANT:`, otherwise, place it below the other `IMPORTANT` entries. If
+you're not sure if your change merits a news entry, or if it's important,
+please ask.
+
+### Coding Style
+
+In general: Just look at the existing source code!
+
+We try to target experienced JavaScript developers who do not necessarily need
+to have a good understanding of Vimperator's source code, nor necessarily
+understand in-depth concepts of other languages like Lisp or Python. Therefore,
+the coding style should feel natural to any JavaScript developer. Of course,
+this does not mean, you have to avoid all new JavaScript features like list
+comprehension or generators. Use them, when they make sense, but don't use them
+when the resulting code is hard to read.
```

Comment:
Can we add a note below this paragraph (perhaps in bold or italics) to only use standards compliant javascript?